### PR TITLE
[RFR] Add rich text input toolbar buttons

### PR DESCRIPTION
**Pull request by bladx**
As discussed in https://github.com/marmelab/react-admin/issues/7806, here is the PR to have both ColorButtons (including font and highlight color) and ImageButtons components for RichTextInput toolbar.

This is my first PR ever on any non-private repository, so please feel free to make me some feedbacks (english or french).

Plus, I noticed that the translation of all buttons (existing ones as well) do not exist, so even in french the titles are in english (default).

### CODE REVIEW

**Comment by bladx** [comment]
I also just noticed that the toolbar is not responsive (missing the flex-wrap so the screen has an horizontal scroll bar).

File: packages/ra-input-rich-text/src/buttons/ColorButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ * Hook that listens clicks outside of the passed ref
+ */
+const useOutsideListener = (
+    ref: React.MutableRefObject<any>,
+    onClick: () => void
+) => {
+    React.useEffect(() => {
+        /**
+         * Alert if clicked on outside of element
+         */
```
Comment (slax57) [inline packages/ra-input-rich-text/src/buttons/ColorButtons.tsx:34]: This comment seems out of place, I think you can simply remove it

**Review by slax57 — CHANGES_REQUESTED** [review]
Otherwise this is a really great contribution, thanks! :pray:

**Comment by bladx** [comment]
Thanks for the kind feedback :)

**Comment by djhi** [comment]
> Plus, I noticed that the translation of all buttons (existing ones as well) do not exist, so even in french the titles are in english (default).

Yes, thank you for this reminder. We're going to include english and french translations directly inside this package.

File: packages/ra-input-rich-text/src/buttons/ColorButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                            onClick={() => selectColor(color[shade])}
+                        ></Box>
+                    ))}
+                </Box>
+            ))}
+        </Card>
+    );
+};
+
+export const ColorButtons = (props: Omit<ToggleButtonProps, 'value'>) => {
```
Comment (djhi) [inline packages/ra-input-rich-text/src/buttons/ColorButtons.tsx:138]: Would you mind moving this component at the top of the file ? We usually put first the most important things.

**Review by djhi — CHANGES_REQUESTED** [review]
Thanks a lot!

**Review by slax57 — APPROVED** [review]

**Comment by fzaninotto** [comment]
Could you please fix the linter warnings?

**Comment by bladx** [comment]
> Could you please fix the linter warnings?

Sorry I introduced them in my last code review commit and did not pay attention.
This should be alright now.

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**